### ISSUE TITLE
Weapon skill-ups

### ISSUE THREAD

**Issue by Stump876** [chromiecraft/chromiecraft#507]
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [x] Horde

##### CONTENT PHASE:

- [x] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

I think the chance for skill up is accurate, it's almost 100% at lower levels and gets increasingly harder the closer to the skill cap of the level you get. I believe that part works. However, It seems like only connecting attacks will give a chance for skill up, I had a hard time seeing which ones I got credit for. I probably should have checked it on a slow 2H weapon as opposed to a 1h axe. Being a higher level would help, I would have more swings to test against. But I don't think it's ticking like it should.

Additionally attacks made on critters (rats, rabbits, snakes, etc...) do not count. 

##### EXPECTED BLIZZLIKE BEHAVIOUR:

[https://classic.wowhead.com/guides/classic-wow-weapon-skills](https://classic.wowhead.com/guides/classic-wow-weapon-skills)
[https://wowwiki-archive.fandom.com/wiki/Weapon_skill](https://wowwiki-archive.fandom.com/wiki/Weapon_skill)
According to wowhead "Each attack that is not evaded has a chance to increase weapon skill" So any block, miss, dodge, parry, or hit should give a chance at skill up. This should include also attacks on critters (rats, rabbits, snakes, etc...) as this is still an attack on something killable.

I should also note that training dummies will not give weapon skill ups as of 3.3.3 Prior to that any skill used (yellow damage) had a chance to level it, but not auto attacks (white damage). [https://www.wowhead.com/npc=32667/training-dummy](https://www.wowhead.com/npc=32667/training-dummy)

##### STEPS TO REPRODUCE THE PROBLEM:

1. .setskill of a weapon to 1 
2. Attack a same level mob with that weapon to level it, each swing (block, miss, dodge, parry) should count a 1 skill point initially and get less frequent the closer to the skill cap of the level you get.

##### EXTRA NOTES:
--

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/91081f4ad8fed98d7bdc9c0ae13d05c86032df9f

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

**Comment by Stump876** [comment]
Doing some more testing it seems like misses do count. However, dodged and parried attacks to not.

For some clarification, the training dummies work properly. I didn't intend to indicate that was an issue, just a pre-emptive test.

**Comment by necropola** [comment]
@Stump876 Just a small tip, if you want to skill up fast. The chance for skillups also depends on the level difference between your character and the target. Funny enough the chance for a skillup increases against lower level targets. The optimal target choice, especially when close to the cap, is a grey target exactly 2 levels below green (1 skill point per hit).

I don't know, if this behavior is intended, but it sure helps to know this little "trick". I personally would prefer to have `AlwaysMaxSkillForLevel = 1` in `worldserver.conf` since I think that skilling weapon proficiencies is just an annoyance which adds nothing to the game experience.

**Comment by Stump876** [comment]
That's interesting, I don't think it's intended. From everything I've read it shouldn't matter what level the mob is. I've also found  Intellect increases the skill up chances. 

I was glad when it was removed from the game in Cataclysm, it just seemed like an unnecessary grind. But I liked getting the achievement for grinding it. Leveling them isn't too bad if you keep on it every few levels. Its more difficult at higher exp rates, or with heirlooms where you have one weapon pretty much the entire time. I can see this becoming more of an irritation at higher levels starting at 1. Leveling something from 1-400 at 80 is a nightmare. But I was going for blizzlike. Again, I think the current "chance to skill-up" script is correct for hits and misses, but dodge and parry don't seem to be included, as well any attacks on critters.
